connman/plugins/sailfish_ofono.c

```diff
@@ -940,17 +948,27 @@ static void connctx_settings_changed(OfonoConnCtx *connctx, void *arg)
 {
 	struct modem_data *md = arg;
 	bool disconnecting = connman_network_get_disconnecting(md->network);
+	int index;
 
 	DBG("index %d ipv4 %p ipv6 %p",md->network ?
 				connman_network_get_index(md->network) : -1,
 			md->connctx->settings, md->connctx->ipv6_settings);
+	DBG("connctx %p md->connctx %p", connctx, md->connctx);
 
 	if (disconnecting) {
 		DBG("network %p disconnecting, skip modem conf", md->network);
 		return;
 	}
 
-	modem_configure(arg);
+	index = modem_configure(arg);
```

Comment:
No, `modem_configure()` apparently does not do anything if `md->network` is `NULL` as the service is also then `NULL`. Hm, it still returns the `index` and I think - provided I remember correctly - my idea was not to set the index in that case. This plugin might need some overhaul and cleanup but I did not see this big work as a part of that because of the focus on CLAT.